setuptools/discovery.py

```diff
@@ -57,12 +57,12 @@
 from distutils import log
 from distutils.util import convert_path
 
-StrIter = Iterator[str]
-
-chain_iter = itertools.chain.from_iterable
-
 if TYPE_CHECKING:
     from setuptools import Distribution
+    from typing_extensions import TypeAlias
+
+StrIter: TypeAlias = Iterator[str]
```

Comment:
I don't feel like this alias brings anything of value:
- It doesn't work around any runtime issue
- It doesn't abstract away any complexity
- It's not clearer than the type it's aliasing, or doesn't help explain it
- It's barely any more concise (because it's using a shorthand instead of being clear and explicit)

**But** it's not private. Could we maybe deprecate it ?